USSF-ORBIT/ussf-portal-cms
# feat: Add LandingPage articles to search results

# SC-3010



## Proposed changes
- Return articles with a category of `LandingPage` in search results
- `LandingPage` articles are given the proper permalink to link them to their associated landing page
- Articles with a category of `ORBITBlog` are also now returned in search results


---



## Reviewer notes
To test:
- Create a Tag
- Create an article with the category `LandingPage` that uses the tag created in the last step
- Create a Landing Page that uses the same tag
- Once everything is published, navigate to the portal and search for your article. It should appear in the search results and have a path that relates it to the landing page. Example path: `http://localhost:3000/landing/test-page/test-article`

Something to be aware of: if an article is linked to more than one landing page, only the first connection that is found will be returned in the search results.

E2E tests: https://github.com/USSF-ORBIT/ussf-portal/pull/357


## Setup



### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-client
yarn dev
```

Login to the cms http://localhost:3001

---

## Code review steps

### As the original developer, I have

- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

## Screenshots
